File: Napakalaki-Java/src/NapakalakiGame/Treasure.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author Cristina
 */
public class Treasure {
    private String name;
    private int bonus;
    private TreasureKind type;
    
    public Treasure(String n, int bonus1, TreasureKind t){
        name=n;
        bonus=bonus1;
        type=t;
    }
    
    public String getName(){
        return name;
    }
    
    public int getBonus(){
        return bonus;
    }
    
    public TreasureKind getType(){
        return type;
    }
    public String toString(){
        return name + " Type: " + type.toString() + " Bonus: +" + Integer.toString(bonus);
    }
}

File: Napakalaki-Ruby/lib/treasure.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module NapakalakiGame

class Treasure
  def initialize(n,bons,tK)
    @name=n
    @bonus=bons
    @tKind=tK
  end
  
  def getName
    @name
  end
  def getBonus
    @bonus
  end
  def getType
    @tKind
  end
  
  def to_s
      "#{@name}, bonus = #{@bonus}, tipo = \n #{@tKind.to_s}\n}"
  end
end
end
